In [1]:
import socket

In [13]:
mysock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
mysock.connect(('www.pythonlearn.com',80))

In [14]:
#http://stackoverflow.com/questions/2411864/python-socket-send-buffer-vs-str
mysock.send(b'GET http://www.pythonlearn.com/code/intro-short.txt HTTP/1.0\n\n')

62

In [15]:
while True:
    data = mysock.recv(512).strip()
    if(len(data)<1):
        break
    print(data)
mysock.close()

b'HTTP/1.1 200 OK\r\nDate: Wed, 18 Nov 2015 14:40:34 GMT\r\nServer: Apache\r\nLast-Modified: Mon, 12 Oct 2015 14:55:29 GMT\r\nETag: "20f7401b-1d3-521e9853a392b"\r\nAccept-Ranges: bytes\r\nContent-Length: 467\r\nCache-Control: max-age=604800, public\r\nAccess-Control-Allow-Origin: *\r\nAccess-Control-Allow-Headers: origin, x-requested-with, content-type\r\nAccess-Control-Allow-Methods: GET\r\nConnection: close\r\nContent-Type: text/plain\r\n\r\nWhy should you learn to write programs?\n\nWriting programs (or programming) is a very creative'
b'and rewarding activity.  You can write programs for \nmany reasons, ranging from making your living to solving\na difficult data analysis problem to having fun to helping\nsomeone else solve a problem.  This book assumes that \neveryone needs to know how to program, and that once \nyou know how to program you will figure out what you want \nto do with your newfound skills.'


b'HTTP/1.1 200 OK\r\nDate: Wed, 18 Nov 2015 14:30:08 GMT\r\n
Server: Apache\r\n
Last-Modified: Mon, 12 Oct 2015 14:55:29 GMT\r\n
ETag: "20f7401b-1d3-521e9853a392b"\r\n

Accept-Ranges: bytes\r\n
Content-Length: 467\r\n
Cache-Control: max-age=604800, public\r\n
Access-Control-Allow-Origin: *\r\n
Access-Control-Allow-Headers: origin, x-requested-with, content-type\r\nAccess-Control-Allow-Methods: GET\r\n
Connection: close\r\n
Content-Type: text/plain\r\



n\r\nWhy should you learn to write programs?\n\nWriting programs (or programming) is a very creative '
b'\nand rewarding activity.  You can write programs for \nmany reasons, ranging from making your living to solving\na difficult data analysis problem to having fun to helping\nsomeone else solve a problem.  This book assumes that \neveryone needs to know how to program, and that once \nyou know how to program you will figure out what you want \nto do with your newfound skills.  \n'
